src/jit/valuenum.cpp

```diff
@@ -1613,6 +1613,7 @@ double ValueNumStore::GetConstantDouble(ValueNum argVN)
 {
     assert(IsVNConstant(argVN));
     var_types argVNtyp = TypeOfVN(argVN);
+    assert(varTypeIsFloating(argVNtyp));
```

Comment:
@tannergooding, your reasoning sounds good to me; I'd say go ahead with this and the other instances.